libgadget/forcetree.h

```diff
@@ -37,64 +37,77 @@ extern struct NODE
         d;
     }
     u;
-}
-*Nodes_base,			/*!< points to the actual memory allocated for the nodes */
-    *Nodes;			/*!< this is a pointer used to access the nodes which is shifted such that Nodes[RootNode]
-                      gives the first allocated node */
+};
 
-/*Structure containing the Node pointer, and the first and last entries*/
-struct TreeBuilder {
-    /*Index of first internal node*/
+/*Structure containing the Node pointer, and various Tree metadata.*/
+/*The node index is an integer with unusual properties:
+ * no = 0..OctTree.firstnode  corresponds to a particle.
+ * no = OctTree.firstnode..OctTree.lastnode corresponds to actual tree nodes,
+ * and is the only memory allocated in OctTree.Nodes_base. After the tree is built this becomes
+ * no = OctTree.firstnode..OctTree.numnodes which is the only allocated memory.
+ * no > OctTree.lastnode means a pseudo particle on another processor*/
+struct OctTree {
+    /*Is 1 if the tree is allocated*/
+    int tree_allocated_flag;
+    /*Index of first internal node. Difference between Nodes and Nodes_base. == MaxPart*/
```

Comment:
Are there any members used only during treebuild? If so, good idea to mark them, since it appears you have merged the tree builder interface with the tree.

Reply:
Done!